src/sdk/service/iam-service.ts

```diff
@@ -151,4 +157,97 @@ export class IamService {
     }
     return false;
   }
+
+  /**
+   * Validate a role creation request.
+   * @param {RoleCreationRequest} roleCreationRequest
+   * @param {CompanyInventory} companyInventory
+   * @returns {Array<Error>}
+   */
+  static validateRole(roleCreationRequest: RoleCreationRequest, companyInventory: CompanyInventory) {
+    const errors: Array<Error> = [];
+    if (roleCreationRequest.policies.length === 0) {
+      errors.push((new Error('A role must have at least one policy.')));
+    }
+    for (const policy of roleCreationRequest.policies) {
+      const entityOptional: InventoryEntity | undefined = companyInventory.getEntityByUuid(policy.entityUuid);
+      if (entityOptional === undefined) {
+        errors.push((new Error('Entity not found in this company.')));
+      } else {
+        if (policy.entityDomain !== entityOptional.type) {
+          errors.push((new Error('The entity is assigned in policy but not for the good domain.')));
```

Comment:
change to "Policy for entity '${?}' has domain type '${?}' but entity is actually of type '${?}'".